src/System.Net.Http/src/System/Net/Http/HPack/HeaderField.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace System.Net.Http.HPack
+{
+    internal struct HeaderField
+    {
+        // http://httpwg.org/specs/rfc7541.html#rfc.section.4.1
+        public const int RfcOverhead = 32;
+
+        public HeaderField(Span<byte> name, Span<byte> value)
+        {
+            Name = new byte[name.Length];
```

Comment:
Ouch.  When is a HeaderField constructed?  Seems like we could end up with a lot of duplicated allocations here, or is the HeaderField itself just constructed once per header and then cached for all future use?

Reply:
I think this code overall needs some perf attention, particularly with regard to allocations. I'm going to defer that for now.